app/src/main/java/com/epfl/drawyourpath/utils/Utils.kt

```diff
@@ -392,4 +394,118 @@ object Utils {
     fun coordinatesToBitmap(coordinates: List<LatLng>, size: Int = 100, paint: Paint = defaultPaint): Bitmap {
         return strokesToBitmap(listOf(coordinatesToStroke(coordinates)), size, paint)
     }
+
+    /**
+     * Reduces the number of points in a Path.
+     * @param path The path we want to reduce
+     * @param maxError The max error percentage relative to the path distance
+     * @return A similar Path with the least useful points removed.
+     */
+    fun reducePath(path: Path, maxError: Float = 0.01F): Path {
+        val reducedPoints = mutableListOf(listOf<LatLng>())
+        if (path.size() <= 2) {
+            return Path(path.getPoints())
+        }
+
+        val epsilon = maxError * path.getDistance().toFloat()
+        for (section in path.getPoints()) {
+            reducedPoints.add(reduceSection(section, epsilon))
+        }
+        reducedPoints.removeFirst()
+        return Path(reducedPoints.toList())
+    }
+
+    /**
+     * Reduces the number of points in a section.
+     * @param section The section we want to reduce
+     * @param epsilon The min distance to keep the point on the section
+     * @return A similar segment with the least useful points removed.
+     */
+    fun reduceSection(section: List<LatLng>, epsilon: Float): List<LatLng> {
+        val reducedPointList = mutableListOf<LatLng>()
+        if (section.size <= 2) {
+            reducedPointList.addAll(section)
+            return reducedPointList.toList()
+        }
```

Comment:
Same here, you can probably just return `section`
```suggestion
        if (section.size <= 2) {
            return section
        }
```

Reply:
Same here, I was making a copy but it is not required